server/tserver/src/main/java/org/apache/accumulo/tserver/tablet/Tablet.java

```diff
@@ -586,6 +589,10 @@ private LookupResult lookup(SortedKeyValueIterator<Key,Value> mmfi, List<Range>
           mmfi.next();
         }
 
+      } catch (ScanYieldException sye) {
+        log.debug("Scan yield exception detected at position " + sye.getPosition());
+        addUnfinishedRange(lookupResult, range, sye.getPosition(), false);
```

Comment:
What happens to data which is buffered to be sent back to the client, but has not yet been sent? (the buffer controlled by the tserver scan max memory property). I think the client would see this data, but I wanted to dbl check with you as you're more familiar than I am now :)

Specifically, say we have a buffer which can hold 500 Key-Values. An Iterator gets through 250 K-V's, and then throws this exception. We mark the `Range` as unfinished, but does the client see those 250 K-V's that were produced by the Iterator?

It seems like `addUnfinishedRange(..)` is then determining for us if we need to do any more work on the current range (e.g. handling the case where we "timed out" after computing the last k-v for the Range).